File: Template/TypeScript/src/Barista.ts
import Tea from './beverage/Tea';
import Coffee from './beverage/Coffee';
import TeaWithHook from './beverage/TeaWithHook';
import CoffeeWithHook from './beverage/CoffeeWithHook';

const main = () => {
  const tea: Tea = new Tea();
  const coffee: Coffee = new Coffee();

  console.log('\nMaking tea...');
  tea.prepareRecipe();

  console.log('\nMaking coffee...');
  coffee.prepareRecipe();

  const teaHook: TeaWithHook = new TeaWithHook();
  const coffeeHook: CoffeeWithHook = new CoffeeWithHook();

  console.log('\nMaking tea...');
  teaHook.prepareRecipe();
  console.log('\nMaking coffee...');
  coffeeHook.prepareRecipe();
};

main();

File: Template/Java/src/Barista.java
import beverage.Tea;
import beverage.TeaWithHook;
import beverage.Coffee;
import beverage.CoffeeWithHook;

public class Barista {
  public static void main(String[] args){
    Tea tea = new Tea();
    Coffee coffee = new Coffee();

    System.out.println("\nMaking tea...");
    tea.prepareRecipe();

    System.out.println("\nMaking coffee...");
    coffee.prepareRecipe();

    TeaWithHook teaHook = new TeaWithHook();
		CoffeeWithHook coffeeHook = new CoffeeWithHook();

		System.out.println("\nMaking tea...");
		teaHook.prepareRecipe();

		System.out.println("\nMaking coffee...");
		coffeeHook.prepareRecipe();
  }
}
